### PR TITLE
Basic functionalities for the Search Bar have been created.

### PR DESCRIPTION
**Pull request by tinchonapa**
Currently the search bar allows the user to input text and dynamically search through all the information that each pickup holds.

It's missing to work along side with the Day filter functionality. As of this moment it is one or the other. They can't be used at the same time.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -129,11 +67,9 @@ const FilterByDayOfWeek = ({ filterByDay, setFilterByDay }) => {
 
   return (
       <>
-        <hr className='horizontal-line'/>
-        <section className="is-horizontal-scrollable centered">
+        <div class="control">
```
Comment (weirdestnerd) [inline src/App.js:70]: Could you revert this change? It's causing the search bar and the list of days to not show up correctly on mobile

https://user-images.githubusercontent.com/13685735/109370462-a7f3da80-785d-11eb-9613-1ead3b4657bf.mov

File: src/App.js [hunk 2]
```diff
@@ -330,7 +286,19 @@ const App = () => {
   const renderPickups = (filterByDay) => {
     const today = WEEKDAYS[DateTime.local().weekday]
     const tomorrow = WEEKDAYS[DateTime.local().plus({ days: 1 }).weekday]
-
+    if (filterByDay && searchTerm) {
+      return (
+        <>
+            <HeadingBanner text={`Showing pickups for ${filterByDay}`}/>
```
Comment (weirdestnerd) [inline src/App.js:292]:
```suggestion
            <HeadingBanner text={`Showing pickups for ${searchTerm} on ${filterByDay}`}/>
```

File: src/App.js [hunk 3]
```diff
@@ -315,10 +266,15 @@ const App = () => {
...
 
-  const renderPickupsByDayOfTheWeek = (weekday) => {
-    const pickupsHappeningOnDay = PICKUPS.filter(pickup => (pickup.day === weekday) && !pickup.hidden)
-    if (pickupsHappeningOnDay.length === 0) return <p>No pickups</p>
-    return pickupsHappeningOnDay.map(pickup => <PickupCard field={pickup.field} address={pickup.address} time={pickup.time} day={pickup.day} contact={pickup.contact}/>)
+  const renderPickupsByDayOfTheWeek = (weekday, search) => {
+    if (weekday && search) { 
+      const pickupsHappeningOnDay = searchResults.filter(pickup => (pickup.day === weekday) && !pickup.hidden);
+      return pickupsHappeningOnDay.map(pickup => <PickupCard field={pickup.field} address={pickup.address} time={pickup.time} day={pickup.day} contact={pickup.contact}/>)
+    } else {
```
Comment (weirdestnerd) [inline src/App.js:273]: You can move this logic into its own function as `renderSearchResultsByDayOfWeek`.

**Review by weirdestnerd — CHANGES_REQUESTED** [review]
This is a great job so far! And it works!!! I like the approach with using `useEffect`. Another way to achieve the same thing would be to move the implement inside of `useEffect` to the `handleChange` function. Either way I'm fine with what you have.

A couple notes I have with this PR are:
- Try to create a PR for one task only. This PR also includes moving components to their files.
- Adding screenshots or screen capture whenever you make UI changes are good practice for PR. It helps the reviewer visualise the changes you made.

I'm requesting changes because the filter for days don't show up fully on mobile. "Monday" is cut off.

**Comment by tinchonapa** [comment]
@weirdestnerd when you get a chance take a look at the code. I finalized the clean-up of the code and fixed the error that we went through yesterday. Inclduing the creation of a new fn for the _renderSearchResultsByDayOfTheWeek_ and making the useEffect more readable by creating a variable in the For In Loop.

File: src/App.js [hunk 4]
```diff
@@ -301,6 +228,21 @@ const App = () => {
...
+  const [searchResults, setSearchResults] = useState([]);
+  const handleSearchTermChange = event => {
+    setSearchTerm(event.target.value);
+  };
+  useEffect(() => {
+    let results = PICKUPS.filter(pickup => {
+      // we want to return true when at least one attribute match
+      for (let property in pickup) {
+        var pickupData = pickup[property];
+        if ((typeof pickupData !== 'boolean') && (pickupData.toLowerCase().includes(searchTerm.toLowerCase()))) return true;
```
Comment (weirdestnerd) [inline src/App.js:241]: I was hoping for something like:
```suggestion
        let attribute = pickup[property]
        let attributeIsSubsetOfSearchTerm = attribute.toLowerCase().includes(searchTerm.toLowerCase())
        if ((typeof attribute !== 'boolean') && (attributeIsSubsetOfSearchTerm)) return true;
```
But I won't block the merge on this.

File: src/App.js [hunk 5]
```diff
@@ -315,8 +257,14 @@ const App = () => {
...
 
-  const renderPickupsByDayOfTheWeek = (weekday) => {
-    const pickupsHappeningOnDay = PICKUPS.filter(pickup => (pickup.day === weekday) && !pickup.hidden)
+  const renderSearchResultsByDayOfTheWeek = (weekday, search) => {
+    const pickupSearchHappeningOnDay = searchResults.filter(pickup => (pickup.day === weekday) && !pickup.hidden);
+    if (pickupSearchHappeningOnDay.length === 0) return <p>No pickups</p>
+    return pickupSearchHappeningOnDay.map(pickup => <PickupCard field={pickup.field} address={pickup.address} time={pickup.time} day={pickup.day} contact={pickup.contact}/>)
+  }
+  
+  const renderPickupsByDayOfTheWeek = (weekday, search) => {
```
Comment (weirdestnerd) [inline src/App.js:266]:
```suggestion
  const renderPickupsByDayOfTheWeek = (weekday) => {
```

**Review by weirdestnerd — APPROVED** [review]

File: src/App.js (hunk 4 above)
Comment (tinchonapa) [reply src/App.js:241]: Makes sense!  I'll add that modification so the code can have better quality.
